simulariumio/data_objects/unit_data.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import logging
+from pint import UnitRegistry
+
+###############################################################################
+
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+
+class UnitData:
+    magnitude: float
+    name: str
+
+    def __init__(
+        self,
+        name: str,
+        magnitude: float = 1.0,
+    ):
+        """
+        This object contains data for units
+
+        Parameters
+        ----------
+        name: str
+            unit name for values (we support this list
+            https://github.com/hgrecco/pint/blob/master/pint/default_en.txt)
+        magnitude: float (optional)
+            multiplier for values (in case they are not given in whole units)
+            Default: 1.0
+        """
+        # standardize and simplify units with pint
+        ureg = UnitRegistry()
+        self._quantity = magnitude * ureg(name)
+        self._quantity = self._quantity.to_compact()
+        self.magnitude = self._clamp_precision(self._quantity.magnitude)
+        self.name = str(self._quantity.units)
```

Comment:
Is there a reason to return the unabbreviated unit name here ("seconds" as opposed to "s")? We only display abbreviated units in the frontend currently, so it would be nice to be able to use the abbreviated string unit straight from the data.

Reply:
pint outputs unit strings unabbreviated, I can look into getting it to output the abbreviated units, but won't you be using a units library anyway?

Reply:
I thought the utility of using a units library would be in screening out nonstandard units, which is being done here with pint. @toloudis and I [talked about this briefly on Slack](https://allencellscience.slack.com/archives/CFT511CE4/p1613161719045500), let me know if I'm missing something.

Would this work? `self.name = f'{self._quantity.units:~}'`
source: https://pint.readthedocs.io/en/0.10.1/tutorial.html#string-formatting

Reply:
well you can't count on all data coming through this package, some people will write their own scripts to convert their data. So I think you still need to check that on the FE.

I changed everything to abbreviations though, that was my preference too just was being lazy :) however, in doing that I discovered that pint's abbreviation for micrometers is wrong 😱 it's "µ" rather than "µm". I wrote a workaround but now I'm wondering if we should put in a bug to pint for that. Seems like someone else would have noticed by now??

Reply:
Ah, ok, I didn't consider people converting data on their own. Thanks for abbreviating the units anyway :) 

What? I just Googled and found that µ is the abbreviation for "micron" before micron got replaced with micrometers..... you probably knew that, but anyway sounds like they need to update their stuff!

Reply:
To me, the unit representation is an identifier that can be used to look up the name of the unit, the order of magnitude of the base SI unit (i.e. 10^6 etc) and other info.  If that identifier happens to be a human readable string abbreviation, that can be output in the front end, that's great!  As long as it uniquely identifies the unit!

For the micro symbol, I wonder if we need to utf-8 encode it for json output or does  pint handle that.